backtracking/rat_in_maze.py

```diff
@@ -4,101 +4,84 @@
 def solve_maze(maze: list[list[int]]) -> bool:
     """
     This method solves the "rat in maze" problem.
-    In this problem we have some n by n matrix, a start point and an end point.
-    We want to go from the start to the end. In this matrix zeroes represent walls
-    and ones paths we can use.
-    Parameters :
-        maze(2D matrix) : maze
+    In this problem, we have an n by n matrix, a start point, and an end point.
+    We want to go from the start to the end. In this matrix, ones represent walls,
+    and zeros represent paths we can use.
+
+    Parameters:
+        maze (2D matrix): The maze where 1 represents walls, and 0 represents paths.
+
     Returns:
-        Return: True if the maze has a solution or False if it does not.
+        bool: True if a solution exists, False otherwise.
+
     >>> maze = [[0, 1, 0, 1, 1],
     ...         [0, 0, 0, 0, 0],
     ...         [1, 0, 1, 0, 1],
     ...         [0, 0, 1, 0, 0],
     ...         [1, 0, 0, 1, 0]]
     >>> solve_maze(maze)
+    True
+    Path:
```

Comment:
Changing this code is modifying what the pytest is looking for.  As discussed in `CONTRIBUTING.md` you can run these tests on your local computer by doing.
% `python3 -m doctest -v backtracking/rat_in_maze.py`